en, zh: Add pd nslookup issue (#1611)
pingcap/docs-tidb-operator: en/deploy-failures.md

root soft stack 10240
```

> ### PD Pod nslookup domain failed

Reviewer: ```suggestion
### PD Pod `nslookup domain failed`
```

You should see some log of PD Pod like:

```
Thu Jan 13 14:55:52 IST 2022
;; Got recursion not available from 10.43.0.10, trying next server
;; Got recursion not available from 10.43.0.10, trying next server
;; Got recursion not available from 10.43.0.10, trying next server
Server: 10.43.0.10
Address: 10.43.0.10#53

** server can't find basic-pd-0.basic-pd-peer.default.svc: NXDOMAIN

nslookup domain basic-pd-0.basic-pd-peer.default.svc failed
```

> This happen when:

Reviewer: ```suggestion
This type of failure occurs in the following situations:
```

Author: 这里下面 "There are two `nameserver` in `/etc/resolv.conf`, and the second one is not IP of CoreDNS."  跟 “he version of PD is:” 必须同时满足

> This type of failure occurs in the following situations:

Reviewer: ```suggestion
This type of failure occurs when the cluster meets both of the following two conditions:
```

> - There two `nameserver` in `/etc/resolv.conf` and the second one is not IP of CoreDNS.

Reviewer: ```suggestion
- There are two `nameserver` in `/etc/resolv.conf`, and the second one is not IP of CoreDNS.
```

> - Version of PD is:

Reviewer: ```suggestion
- The version of PD is:
```

>     - Greater than or equals v5.0.5.
>     - Greater than or equals v5.1.4.
>     - Greater than or equals v5.2.4.

Reviewer: ```suggestion
    - Greater than or equal to v5.0.5.
    - Greater than or equal to v5.1.4.
    - Greater than or equal to v5.2.4.
```

    - All 5.3 versions.

> To address this, add `startUpScriptVersion` to TiDBCluster as:

Reviewer: ```suggestion
To address this failure, add `startUpScriptVersion` to TidbCluster as:
```

```yaml
...
spec:
  pd:
    startUpScriptVersion: "v1"
...
```

> The reason is nslookup version is changed in the image, See: [#4379](https://github.com/pingcap/tidb-operator/pull/4379). after adding `startUpScriptVersion` as `v1`, we will use `dig` to check DNS instead of using nslookup.

Reviewer 2: ```suggestion
The reason is that there is something wrong with the nslookup in the base image, see detail in the description of [#4379](https://github.com/pingcap/tidb-operator/pull/4379). After adding `startUpScriptVersion` as `v1`, we will use `dig` to check DNS instead of using nslookup.
```

> The reason is that there is something wrong with the nslookup in the base image, see detail in the description of [#4379](https://github.com/pingcap/tidb-operator/pull/4379). After adding `startUpScriptVersion` as `v1`, we will use `dig` to check DNS instead of using nslookup.

Reviewer: ```suggestion
This failure occurs because there is something wrong with the `nslookup` in the base image (see detail in [#4379](https://github.com/pingcap/tidb-operator/pull/4379)). After configuring `startUpScriptVersion` to `v1`, TiDB Operator uses `dig` to check DNS instead of using `nslookup`.
```
